src/module/vue/components/progress/progress-track.vue

```diff
@@ -1,35 +1,132 @@
 <template>
-  <div class="flexrow track">
-    <div class="flexcol track-box" v-for="(box, i) in boxes" :key="'box' + i">
-      <svg
-        xmlns="http://www.w3.org/2000/svg"
-        viewBox="0 0 100 100"
-        v-html="box"
-      ></svg>
-    </div>
-  </div>
+  <article
+    :tabindex="0"
+    class="progress-track"
+    :class="{
+      ['compact-progress']: compactProgress,
+    }"
+    role="slider"
+    :aria-label="$t('IRONSWORN.PROGRESS.Track')"
+    aria-orientation="horizontal"
+    :data-rank="numericRank"
+    :data-ticks="ticks"
+    :data-score="score"
+    :aria-valuenow="ticks"
+    :aria-valuemin="0"
+    :aria-valuemax="40"
+    :aria-valuetext="`${score} progress (${ticks} ticks)`"
+    :data-tooltip="`${score} progress (${ticks} ticks)`"
+  >
+    <ProgressTrackBox
+      v-for="(boxTicks, i) in boxes"
+      tabindex="-1"
+      role="presentational"
+      :key="`progress-box-${i + 1}`"
+      :ticks="boxTicks ?? 0"
+      :isOverflowBox="legacyOverflow"
+    />
+  </article>
 </template>
 
+<style lang="less">
```

Comment:
Do you have a method for when these are `scoped` and when not? If they're not, they inject styles into the global namespace, and now you have to worry about class-name collisions.

Reply:
i think ive touched on some of my reasoning in previous code reviews. might bear discussing, though, so ive put some thoughts in the relevant discord thread